kong/clustering/data_plane.lua

```diff
@@ -175,55 +170,44 @@ function _M:communicate(premature)
   --                 and is also responsible for handling timeout detection
 
   local ping_immediately
-  local config_exit
   local next_data
```

Comment:
The `next_data` variable does not seem to be used anymore.

Reply:
No, `next_data` is used in `read_thread`.

I also think it is unnecessary, but I won't remove it in this PR.
Perhaps I could create another PR to remove it.

Reply:
As far as I can see, no action is triggered by setting `next_data` to `true`.

Reply:
I think that the purpose of `next_data` is to keep the  config data in case of replaced by "next" reconfiguration action.

if `next_data ~= data`, it means that cp has pushed the new data to dp.
if `next_data == data`, ti means that no new data is pushed to dp, we can set it to `nil` then gc it.

Reply:
`next_data` is never read, except here:

```
    if next_data == data then
      next_data = nil
    end
```

The only thing that happens is that it is set to `nil` when it is equal to `data`.  Why is holding on to the data useful?  How do other references to `data` that may exist not suffice from protecting it from garbage collection?  Is there something non-obvious going on with the memory management?  If so, can this please be clearly described in a comment?

Reply:
In this PR I want to focus on removing `config_thread`, so I try to keep the previous logic.

I also think that `next_data` is meanless, but I will remove it in another PR, Let's do it step by step.